### ISSUE TITLE
Private repository with basic auth

### ISSUE THREAD

**Issue by adrpp** [python-poetry/poetry#4623]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu 21.04 hirsute
- **Poetry version**: Poetry version 1.1.11
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/adrpp/57d68b817e721051220e8a7b3f75fa7f

## Issue

Hello,

first of all, thanks for this great packaging tool :)
I have some questions (and a feeling a part of it might be a bug ?) about using poetry with private repository with basic auth.

My setup is:
- private repo on nexus, mirrors public packages
 - all requests are made to private repository and if it's a company made package, it's provided by  private repository, otherwise it will be provided from 'mirrored pool'
- in my application, as provided in gist link, there is one dependency called "my.dot.notated.dependency" which is provided by private repository

Would be happy with having uri inside of `pyproject.toml`, but the credentials set via env vars or cli (as a part of a pipeline job).
Regarding cli, `poetry config http-basic.nexus username password` should do the trick, however, it doesn't.

Only way it works is to put the creds into repo uri as part of `[[tool.poetry.source]]`.
`poetry config http-basic.nexus username password` will be somehow persisted somewhere, as `poetry config --list` shows related section, but won't get used by poetry / pip while accessing the repository (?)

Setting `poetry config http-basic.nexus username password` but ommiting credentials in the `[[tool.poetry.source]]` section leads to 401 / Unauthorized responses. Looks like resolving dependencies against nexus pypi repository requires repo url contains credentials.

Update:
similar behavior mentioned here: https://github.com/python-poetry/poetry/issues/4389
However, setting `POETRY_HTTP_BASIC_NEXUS_USERNAME` & `POETRY_HTTP_BASIC_NEXUS_PASSWORD` ("NEXUS" due to my setting of repo name, see gist file) doesn't auth the request like mentioned in the issue.

Thanks for any help in advice.
Adrian

**Comment by JamieMcKernanKaizen** [comment]
I am also having this issue. I am using:
`poetry 1.1.11`
`Ubuntu 20.04.3 LTS`

I've observed that when running `poetry config http-basic.repo_name username password` the password _does_ get added to the built in Ubuntu password store (called Passwords and Keys). But it seems poetry can't access this and gives the 401 instead.

The current workaround I've found is to disable the keyring library, and edit the `auth.toml` myself with the username and password.
